File: src/SPS.Domain/Entities/PopupAnnouncement.cs
using SPS.Domain.Common;
using SPS.Domain.Enums;

namespace SPS.Domain.Entities;

/// <summary>
/// 彈窗公告
/// </summary>
public class PopupAnnouncement : BaseEntity<int>
{
    /// <summary>
    /// 標題
    /// </summary>
    public string Title { get; set; } = string.Empty;

    /// <summary>
    /// 內容（HTML）
    /// </summary>
    public string? Content { get; set; }

    /// <summary>
    /// 圖片 ID
    /// </summary>
    public int? ImageId { get; set; }

    /// <summary>
    /// 點擊跳轉連結
    /// </summary>
    public string? LinkUrl { get; set; }

    /// <summary>
    /// 連結開啟方式 (_self / _blank)
    /// </summary>
    public string? LinkTarget { get; set; }

    /// <summary>
    /// 觸發路由（JSON array: ["/", "/products"]）
    /// </summary>
    public string Routes { get; set; } = "[\"/\"]";

    /// <summary>
    /// 顯示頻率
    /// </summary>
    public PopupFrequency Frequency { get; set; } = PopupFrequency.OncePerSession;

    /// <summary>
    /// 優先順序（數字小優先）
    /// </summary>
    public int Priority { get; set; }

    /// <summary>
    /// 生效開始時間
    /// </summary>
    public DateTime? StartDate { get; set; }

    /// <summary>
    /// 生效結束時間
    /// </summary>
    public DateTime? EndDate { get; set; }

    /// <summary>
    /// 是否顯示關閉按鈕
    /// </summary>
    public bool ShowCloseButton { get; set; } = true;

    /// <summary>
    /// 是否顯示「今日不再顯示」選項
    /// </summary>
    public bool ShowDontShowToday { get; set; } = true;

    /// <summary>
    /// 是否啟用
    /// </summary>
    public bool Published { get; set; }

    /// <summary>
    /// 排序
    /// </summary>
    public int Ordinal { get; set; }

    // Navigation properties
    public Picture? Image { get; set; }
}

File: docs/old/types/popup-announcement.ts
/**
 * 彈跳公告
 */
export interface PopupAnnouncement {
    id: number;
    /** 標題 */
    title: string;
    /** 內容 (HTML 或純文字) */
    content: string;
    /** 圖片 ID */
    imageId: number | null;
    /** 圖片 URL */
    imageUrl: string | null;
    /** 連結 URL */
    linkUrl: string | null;
    /** 連結開啟方式 (_self, _blank) */
    linkTarget: string | null;
    /** 適用路由列表 */
    routes: string[];
    /** 顯示頻率 (0: 每次, 1: 每天一次, etc.) */
    frequency: number;
    /** 優先順序 */
    priority: number;
    /** 開始日期 */
    startDate: string;
    /** 結束日期 */
    endDate: string;
    /** 是否顯示關閉按鈕 */
    showCloseButton: boolean;
    /** 是否顯示「今日不再顯示」選項 */
    showDontShowToday: boolean;
    /** 是否發布 */
    published: boolean;
    /** 排序 */
    ordinal: number;
    /** 建立時間 */
    createdTime: string;
    /** 更新時間 */
    updatedTime: string;
}
